### PR TITLE
T18011 JenaUtils.read support jsonld

### PR DESCRIPTION
**Pull request by michielrogissart-cognizone**
See issue #215

### CODE REVIEW

**Review by natan-cox-cognizone — APPROVED** [review]
no deprecation?

File: cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java [hunk 1]
```diff
@@ -254,6 +254,10 @@ private static String getRdfSyntax(org.springframework.core.io.Resource resource
       // .ttl is extension for text/turtle
       return "TURTLE";
     }
+    if("jsonld".equals(extension)) {
+      // .jsonld is extension for application/ld+json
+      return "JSONLD";
+    }
```
Comment (domivds) [inline cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java]: can't we put this in a map?

File: cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java [hunk 2]
```diff
@@ -237,28 +237,21 @@ private static RDFReader getReaderByRdfSyntax(Model model, String language) {
...
-    }
-    if ("ttl".equals(extension)) {
-      // .ttl is extension for text/turtle
-      return "TURTLE";
-    }
+    Map<String, String> extensionToLanguageCode = getExtensionToLanguageMap();
 
-    // any other extension falls back to RDF XML
-    return null;
+    return extensionToLanguageCode.getOrDefault(extension, null);
```
Comment (domivds) [inline cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java]: just get is ok, if key is not present, null is returned - add a comment that fallback (and null) is RDF/XML

File: cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java [hunk 3]
```diff
@@ -237,28 +237,21 @@ private static RDFReader getReaderByRdfSyntax(Model model, String language) {
...
+  private static Map<String, String> getExtensionToLanguageMap() {
+    Map<String, String> result = new HashMap<>();
+
+    result.put("nt", "N-TRIPLE");
+    result.put("n3", "N3");
+    result.put("ttl", "TURTLE");
+    result.put("jsonld", "JSONLD");
+
+    return result;
+  }
```
Comment (domivds) [inline cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java]: private static final Map<> in class and init in static initializer (make map synchronized)

Comment (domivds) [inline cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java]: missing empty line after method

File: cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java [hunk 4]
```diff
@@ -237,26 +246,9 @@ private static RDFReader getReaderByRdfSyntax(Model model, String language) {
...
-    }
-    if ("ttl".equals(extension)) {
-      // .ttl is extension for text/turtle
-      return "TURTLE";
-    }
 
-    // any other extension falls back to RDF XML
-    return null;
+    // when return value is null, fall back to RDF/XML
+    return extensionToLanguageMap.getOrDefault(extension, null);
```
Comment (domivds) [inline cogni-libs/src/main/java/zone/cogni/libs/jena/utils/JenaUtils.java:251]: it's ok to do this as here the default return value is null
extensionToLanguageMap.get(extension)

**Review by domivds — APPROVED** [review]

**Pull request merged**